microsoft/PowerToys
# Enable Static Analysis on programs plugin

## Summary of the Pull Request

1.  Installs FxCop on Microsoft.Plugin.Program
2.  Treat all warnings as errors on Microsoft.Plugin.Program. 
3. Fixes static analysis warnings. 

## PR Checklist
* [x] Applies to #5129, #2059 

## Info on Pull Request
I apologize for the large number of commits.   The number of fixes was larger than I originally anticipated.  Each commit should compile and run, but hasn't been as thoroughly tested as all of the changes on this branch together.  If desired I can extract some of the commits into separate PRs,  but it would be easiest to test from this branch if running the changes locally.  

## Validation Steps Performed

_How does someone test & validate?_
1.  Verify Packaged applications, installed when Powertoys Run (powerlauncher.exe) is running, are visible.
2.  Verify Packaged applications, uninstalled when Powertoys Run (powerlauncher.exe) is running, are visible.
1.  Verify Win32 applications, installed when Powertoys Run (powerlauncher.exe) is running, are visible.
2.  Verify Win32 applications, uninstalled when Powertoys Run (powerlauncher.exe) is running, are visible.
1.  Verify that icons dynamically update when switching between light and dark themes.
1. Run unit tests
